clipper_admin/clipper_admin/kubernetes/kubernetes_container_manager.py

```diff
@@ -85,103 +119,133 @@ def __init__(self,
+        self.template_engine = jinja2.Environment(
+            loader=jinja2.FileSystemLoader(cur_dir, followlinks=True),
+            undefined=jinja2.StrictUndefined)
+
+        self.logger = ClusterAdapter(logger, {
+            'cluster_name': self.cluster_name
+        })
 
     def start_clipper(self,
                       query_frontend_image,
                       mgmt_frontend_image,
                       cache_size,
                       num_frontend_replicas=1):
+        self._start_redis()
+        self._start_mgmt(mgmt_frontend_image)
         self.num_frontend_replicas = num_frontend_replicas
+        self._start_query(query_frontend_image, cache_size,
+                          num_frontend_replicas)
+        self._start_prometheus()
+        self.connect()
 
+    def _start_redis(self, sleep_time=5):
         # If an existing Redis service isn't provided, start one
         if self.redis_ip is None:
-            name = 'redis'
             with _pass_conflicts():
                 self._k8s_beta.create_namespaced_deployment(
-                    body=yaml.load(
-                        open(
-                            os.path.join(cur_dir,
-                                         '{}-deployment.yaml'.format(name)))),
-                    namespace='default')
+                    body=self._generate_config(
+                        CONFIG_FILES['redis']['deployment'],
+                        cluster_name=self.cluster_name),
+                    namespace=self.namespace)
 
             with _pass_conflicts():
-                body = yaml.load(
-                    open(
-                        os.path.join(cur_dir, '{}-service.yaml'.format(name))))
-                body["spec"]["ports"][0]["port"] = self.redis_port
+                body = self._generate_config(
+                    CONFIG_FILES['redis']['service'],
+                    public_redis_port=self.redis_port,
+                    cluster_name=self.cluster_name)
                 self._k8s_v1.create_namespaced_service(
-                    body=body, namespace='default')
-            time.sleep(10)
+                    body=body, namespace=self.namespace)
+            time.sleep(sleep_time)
 
-        for name, img in zip(['mgmt-frontend', 'query-frontend'],
-                             [mgmt_frontend_image, query_frontend_image]):
+            self.redis_ip = 'redis-at-{cluster_name}'.format(
+                cluster_name=self.cluster_name)
+
+    def _start_mgmt(self, mgmt_image):
+        with _pass_conflicts():
+            mgmt_depolyment_data = self._generate_config(
+                CONFIG_FILES['management']['deployment'],
+                image=mgmt_image,
+                redis_service_host=self.redis_ip,
+                redis_service_port=self.redis_port,
+                cluster_name=self.cluster_name)
+            self._k8s_beta.create_namespaced_deployment(
+                body=mgmt_depolyment_data, namespace='default')
```

Comment:
Missed a 'default' namespace reference. It's definitely worth grepping for "default" in this file to double check you didn't miss any (although I know #500 takes care of this for us).